crates/integration_tests/tests/shared_tests/merge_append_test.rs

```diff
@@ -0,0 +1,206 @@
+    let col3 = BooleanArray::from(vec![Some(true); 100]);
+    let batch = RecordBatch::try_new(schema.clone(), vec![
+        Arc::new(col1) as ArrayRef,
+        Arc::new(col2) as ArrayRef,
+        Arc::new(col3) as ArrayRef,
+    ])
+    .unwrap();
+    data_file_writer.write(batch.clone()).await.unwrap();
+    data_file_writer.close().await.unwrap()
+}
+
+#[tokio::test]
+async fn test_append_data_file() {
+    let fixture = get_shared_containers();
+    let rest_catalog = RestCatalog::new(fixture.catalog_config.clone());
+    let ns = random_ns().await;
+
+    let schema = Schema::builder()
+        .with_schema_id(1)
+        .with_identifier_field_ids(vec![2])
+        .with_fields(vec![
+            NestedField::optional(1, "foo", Type::Primitive(PrimitiveType::String)).into(),
+            NestedField::required(2, "bar", Type::Primitive(PrimitiveType::Int)).into(),
+            NestedField::optional(3, "baz", Type::Primitive(PrimitiveType::Boolean)).into(),
+        ])
+        .build()
+        .unwrap();
+    let table_creation = TableCreation::builder()
+        .name("t1".to_string())
+        .schema(schema.clone())
+        .build();
+    let mut table = rest_catalog
+        .create_table(ns.name(), table_creation)
+        .await
+        .unwrap();
+
+    // Enable merge append for table
+    let tx = Transaction::new(&table);
+    table = tx
+        .set_properties(HashMap::from([
+            (MANIFEST_MERGE_ENABLED.to_string(), "true".to_string()),
+            (MANIFEST_MIN_MERGE_COUNT.to_string(), "4".to_string()),
+            (MANIFEST_TARGET_SIZE_BYTES.to_string(), "7000".to_string()),
+        ]))
+        .unwrap()
+        .commit(&rest_catalog)
+        .await
+        .unwrap();
+
+    // fast append data file 3 time to create 3 manifest
+    let mut original_manifest_entries = vec![];
+    for _ in 0..3 {
+        let data_file = write_new_data_file(&table).await;
+        let tx = Transaction::new(&table);
+        let mut append_action = tx.fast_append(None, vec![]).unwrap();
+        append_action.add_data_files(data_file.clone()).unwrap();
+        let tx = append_action.apply().await.unwrap();
+        table = tx.commit(&rest_catalog).await.unwrap()
+    }
+    let manifest_list = table
+        .metadata()
+        .current_snapshot()
+        .unwrap()
+        .load_manifest_list(table.file_io(), table.metadata())
+        .await
+        .unwrap();
+    assert_eq!(manifest_list.entries().len(), 3);
+
+    // construct test data
+    for (idx, entry) in manifest_list.entries().iter().enumerate() {
+        let manifest = entry.load_manifest(table.file_io()).await.unwrap();
+        assert!(manifest.entries().len() == 1);
+
+        // For this first manifest, it will be pack with the first additional manifest and
+        // the count(2) is less than the min merge count(4), so these two will not merge.
+        // See detail: `MergeManifestProcess::merge_group`
+        if idx == 0 {
```

Comment:
I'm not sure whether it's expected behaviour. This manifest will not be merged, so the status will remain `Added`.